projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-detail-items/order-detail-items.component.ts

```diff
@@ -31,14 +32,18 @@ export class OrderDetailItemsComponent implements OnInit {
   cancel$: Observable<Consignment[]>;
 
   ngOnInit() {
-    this.orderPromotions$ = this.promotionService.getOrderPromotions(
-      this.promotionLocation
-    );
+    this.orderPromotions$ = this.promotionService.getOrderPromotions();
     this.others$ = this.getOtherStatus(...completedValues, ...cancelledValues);
     this.completed$ = this.getExactStatus(completedValues);
     this.cancel$ = this.getExactStatus(cancelledValues);
   }
 
+  getAllOrderEntryPromotions(
```

Comment:
no need this change, call `getProductPromotionForAllEntries(items)` in cart-item-list.